sdk/communication/communication-common/src/identifierModelSerializer.ts

```diff
@@ -160,7 +164,36 @@ export const deserializeCommunicationIdentifier = (
 ): CommunicationIdentifierKind => {
   assertMaximumOneNestedModel(serializedIdentifier);
 
-  const { communicationUser, microsoftTeamsUser, phoneNumber } = serializedIdentifier;
+  const { communicationUser, microsoftTeamsUser, phoneNumber, kind } = serializedIdentifier;
+  if (kind != null) {
+    if (kind === "communicationUser" && communicationUser) {
+      return {
+        kind: "communicationUser",
+        communicationUserId: assertNotNullOrUndefined({ communicationUser }, "id"),
+      };
+    }
+    if (kind === "phoneNumber" && phoneNumber) {
+      return {
+        kind: "phoneNumber",
+        phoneNumber: assertNotNullOrUndefined({ phoneNumber }, "value"),
+        rawId: assertNotNullOrUndefined({ phoneNumber: serializedIdentifier }, "rawId"),
+      };
+    }
+    if (kind === "microsoftTeamsUser" && microsoftTeamsUser) {
+      return {
+        kind: "microsoftTeamsUser",
+        microsoftTeamsUserId: assertNotNullOrUndefined({ microsoftTeamsUser }, "userId"),
+        isAnonymous: assertNotNullOrUndefined({ microsoftTeamsUser }, "isAnonymous"),
+        cloud: assertNotNullOrUndefined({ microsoftTeamsUser }, "cloud"),
+        rawId: assertNotNullOrUndefined({ microsoftTeamsUser: serializedIdentifier }, "rawId"),
+      };
+    }
+    return {
+      kind: "unknown",
+      id: assertNotNullOrUndefined({ unknown: serializedIdentifier }, "rawId"),
+    };
+  }
+
```

Comment:
pls refactor according to https://github.com/Azure/azure-sdk-for-net/pull/31627/files#r1000528967